Applications/Spire/Source/KeyBindings/CancelKeyBindingsTableView.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Spire/KeyBindings/CancelKeyBindingsTableView.hpp"
+#include <QLabel>
+#include <QVBoxLayout>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/KeyBindings/KeySequenceItemDelegate.hpp"
+#include "Spire/Ui/ItemPaddingDelegate.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+using ValidSequence = KeySequenceEditor::ValidKeySequence;
+
+CancelKeyBindingsTableView::CancelKeyBindingsTableView(
+    const std::vector<KeyBindings::CancelActionBinding>& bindings,
+    QWidget* parent)
+    : KeyBindingsTableView(make_fixed_header(parent), parent),
+      m_model(nullptr) {
+  set_key_bindings(bindings);
+  setFixedWidth(scale_width(853));
+  set_width(scale_width(853));
+  set_height(scale_height(376));
+  set_column_width(0, scale_width(238));
+  auto valid_sequences = std::vector<std::vector<std::set<Qt::Key>>>(
+    {ValidSequence({{Qt::Key_Escape}}),
+    ValidSequence({{Qt::Key_Shift, Qt::Key_Alt, Qt::Key_Control},
+    {Qt::Key_Escape}})});
+  auto item_delegate = new KeySequenceItemDelegate(valid_sequences, this);
+  item_delegate->connect_item_modified_signal([=] (auto index) {
+    on_key_sequence_modified(index);
+  });
+  set_column_delegate(1, item_delegate);
+}
+
+void CancelKeyBindingsTableView::set_key_bindings(
+    const std::vector<KeyBindings::CancelActionBinding>& bindings) {
+  m_model = new CancelKeyBindingsTableModel(bindings, this);
```

Comment:
Will Qt release the resources used by the previous model?

Reply:
The documentation claims that the model will be deleted if the parent is the view:

https://doc.qt.io/qt-5/qabstractitemview.html#setModel

Here it claims that the selection model is also deleted if it has a parent:

https://doc.qt.io/qt-5/qabstractitemview.html#setSelectionModel

I'll verify that setting the model's parent also sets the selection model's parent.

Reply:
Nevermind, that is incorrect, while it will delete the old model, it looks like it won't do so until the view itself is deleted.